source/common/router/router.cc

```diff
@@ -935,6 +1060,8 @@ bool Filter::setupRedirect(const Http::HeaderMap& headers) {
 void Filter::doRetry() {
   is_retry_ = true;
   attempt_count_++;
+  ASSERT(pending_retries_ > 0);
+  pending_retries_--;
```

Comment:
How can you have more than one pending retry when there's only one retry callback tracked at any point?

Reply:
i can't, but when implementing the inititial requests part I might.